### ISSUE TITLE
Multiple text columns in Microsoft Word

### ISSUE THREAD

**Issue by gmhtml** [nvaccess/nvda#7921]
Create a document in Microsoft word.
Under page layout select columns and press enter.

Shift Tab over to more columns and press enter.
Choose two, or more by pressing the arrow up or down on the edit spin field, and press enter.
You may tab over to the other options and change them as desired before pressing enter, but there is no feedback just the same at the end.

When the document is scrutinized using arrow keys etc. there is no indication whatsoever that the original layout has changed.

### Expected behavior:
> Tell us what should happen.

It would be very helpful to be able to detect the column changes as one arrows or move through the document by word.
It would be quite useful better yet, to be able to navigate by column, or down a particular  column such as is in the case of a uniform 
table.

Again, at present, there is no indication other than in the status line that multiple columns exist.

These tests have been carried out using an installed version of NVDA 17.4 on Microsoft word 2016, 2010, and 2007, on a windows10 box 64, and 32 bit.

**Comment by derekriemer** [comment]
do you have the Pages box checked in document formatting?

**Comment by fernando-jose-silva** [comment]
My suggestion for this call would be for nvda to provide a shortcut key to switch between the text columns.

**Comment by bdorer** [comment]
Hi Derek, yes you need the page checkbox enabled to have this working. I think this should be noted in the userguide or this option should be reworded to pages/columns.

Navigation is difficult. If you are in the left column, you have to rightarrow to the last sign of this column and rightarrow once again to moove the cursor to the right column. If you press only downArrow to read the Text, the cursor stays in the left columnand leaves it.
I don't know wheather there are word shortcuts to work with columns.

**Comment by gmhtml** [comment]
I can't  get any feedback despite having the pages checked. I also have rows and columns checked though that's for tables. 


Thanks,

frank

**Comment by derekriemer** [comment]
@feerrenrut I remember you working on columns/sections last year, do you have any insight into the issue here?

**Comment by feerrenrut** [comment]
This column detection code was far from perfect, we rely on knowing about and summing all the offsets that could occur. I think there were some cases we could not support.

To track this down, getting access to the document being used would allow us to investigate what formatting is set, and why NVDA is not able to report. Other than that, getting access to a debug level log may include information that explains the problem. The log should at least confirm that the column detection code is running, and there are not any errors etc.

**Comment by Adriani90** [comment]
I can reproduce this issue in a standard word document where formating properties have not been changed.

When Holding CTRL key down and reading by paragraphs, or when reading the whole document by pressing NVDA+a in laptop layout, NVDA reports "continuous section" instead of "text column one, text column two" etc.

However, when reading line by line with arrow keys, NVDA does not Report any column indications.
Navigation by columns is not possible and there is no info that there are more columns when entering the document.

Here is the developer info:

```
INFO - globalCommands.GlobalCommands.script_navigatorObject_devInfo (09:39:52.549):
Developer info for navigator object:
name: u'Microsoft Word-Dokument'
role: ROLE_EDITABLETEXT
states: STATE_MULTILINE, STATE_FOCUSABLE, STATE_FOCUSED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.Dynamic_IAccessibleWordDocumentWindowNVDAObject object at 0x050908D0>
Python class mro: (<class 'NVDAObjects.Dynamic_IAccessibleWordDocumentWindowNVDAObject'>, <class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.winword.WordDocument'>, <class 'NVDAObjects.behaviors.EditableTextWithoutAutoSelectDetection'>, <class 'editableText.EditableTextWithoutAutoSelectDetection'>, <class 'NVDAObjects.behaviors.EditableText'>, <class 'editableText.EditableText'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <type 'object'>)
description: None
location: (165, 162, 1498, 809)
value: None
appModule: <'winword' (appName u'winword', process ID 3044) at address 4eb40f0>
appModule.productName: u'Microsoft Office 2010'
appModule.productVersion: u'14.0.7192.5000'
TextInfo: <class 'NVDAObjects.window.winword.WordDocumentTextInfo'>
windowHandle: 460180
windowClassName: u'_WwG'
windowControlID: 0
windowStyle: 1442840576
windowThreadID: 6128
windowText: u'Microsoft Word-Dokument'
displayText: u''
IAccessibleObject: <POINTER(IAccessible) ptr=0x274f48 at 50cc260>
IAccessibleChildID: 0
IAccessible event parameters: windowHandle=460180, objectID=-4, childID=0
IAccessible accName: u'Microsoft Word-Dokument'
IAccessible accRole: ROLE_SYSTEM_CLIENT

IAccessible accState: STATE_SYSTEM_FOCUSED, STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_VALID (1048580)
IAccessible accDescription: None
IAccessible accValue: None
```

**Comment by feerrenrut** [comment]
When testing this with MS word 2013 and 
[simple-test-columns-nvda.docx](https://github.com/nvaccess/nvda/files/1679841/simple-test-columns-nvda.docx)

Using nvda version: `next-14810,33118eb1` with the "Pages" document formatting options selected, I get the following announced as I read by word with `ctrl+right arrow`:

```
This is a paragraph with only one column continuous section break blank section 2 column 1 of 2 This is a column one paragraph This is the end of the column column 2 of 2 And here is the second column continuous section break blank section 3 column 1 of 1 This is another paragraph with only one column . 
```

There is currently no support for jumping between columns, to get to another column one must continue to move by word. Better yet, edit or read documents in outline or draft view modes. This puts all aspects of the document in a flat structure, allowing you to read each line (even in different columns) just by pressing down arrow.

@Adriani90 in this case developer info doesn't help much, as stated in my last message the things we are after to investigate this one is an example document, and log at debug level logging. Please feel free to use the document I provided to create a log file that demonstrates the problem.

**Comment by Adriani90** [comment]
@Feerrenrut, I am sorry I was not Aware that reporting of columns depends on page Setting in document formating. With pages enabled, columns are properly reported. However, I would split pages and text columns in two Radio Buttons in document formating. When pages is activated, Performance decreases in big MS Word documents. Especially in big tables.

Regarding column Navigation, it would be a useful Feature in documents where data looks like a table but in fact there are text columns. In that case, line by line Navigation would not help because data in the right column could be linked to data in the left column (i.e. table of Contents).

However, in case NV Access would decide to implement this Feature, NVDA must recognize and report when the text Region contains more than one column. This must be reported during text Navigation.

**Comment by SaschaCowley** [comment]
Works when accessing Word via the object model, but not when accessing via UIA.
Tested with NVDA alpha-37521,f29d41fb (2025.3.0.37521) and Microsoft® Word for Microsoft 365 MSO (Version 2506 Build 16.0.18925.20076) 64-bit

**Comment by Bibeksharma5** [comment]
I also agree this. Moreover, when I open any word document having text in multiple columns, NVDA should indicate this. If a document contains text divided into two columns, NVDA should indicate "2 text columns". Actually, I'm a student depending upon laptop for study as well as exams. In exam, teachers used to give me the question paper in docs format dividing the text into two columns. This helps me to understand the division of the document.
